src/caja-desktop-window.c

```diff
@@ -191,11 +225,47 @@ caja_desktop_window_new (CajaApplication *application,
     /* will cause the desktop window to open at the wrong size in gtk 3.20 */
     gtk_window_set_default_size (GTK_WINDOW (window), -1, -1);
 
+    /*For wayland only
+     *Code taken from gtk-layer-shell simple-example.c
+     */
+#ifdef HAVE_WAYLAND
+    if (GDK_IS_WAYLAND_DISPLAY (display))
+    {
+        GtkWindow *gtkwin;
+        gtkwin = (GTK_WINDOW(window));
+
+        /* Before the window is first realized, set it up to be a layer surface */
+        gtk_layer_init_for_window (gtkwin);
+
+        /* Order below normal windows */
+        gtk_layer_set_layer (gtkwin, GTK_LAYER_SHELL_LAYER_BOTTOM);
+
+	    gtk_layer_set_namespace (gtkwin, "desktop");
+
+        /*Anchor the desktop to all four corners
+         *This is much simpler than on x11 and
+         *should always render the desktop across
+         *all of the screen
+         */
+         gtk_layer_set_anchor(gtkwin,GTK_LAYER_SHELL_EDGE_TOP, TRUE);
+         gtk_layer_set_anchor(gtkwin,GTK_LAYER_SHELL_EDGE_BOTTOM, TRUE);
+         gtk_layer_set_anchor(gtkwin,GTK_LAYER_SHELL_EDGE_LEFT, TRUE);
+         gtk_layer_set_anchor(gtkwin,GTK_LAYER_SHELL_EDGE_RIGHT, TRUE);
```

Comment:
These need 8 spaces (they are inside` an if (GDK_IS_WAYLAND_DISPLAY (display))`) clause-but were indented 9 spaces instead of 8, an easy screwup to make after a multiline comment. Fixed again

Reply:
I counted the spaces from the point where `if` begins.
Any way, you fixed it, thx.
Can you please add a space between `gtkwin,` and `GTK_Layer_SHELL_.....`?
Like this for example:
`gtk_layer_set_anchor (gtkwin, GTK_LAYER_SHELL_EDGE_TOP, TRUE);`

Reply:
Done